auth-api/src/auth_api/services/membership.py

```diff
@@ -315,14 +316,29 @@ def deactivate_membership(self, **kwargs):
     @staticmethod
     def _add_or_remove_group(model: MembershipModel):
         """Add or remove the user from/to account holders / product keycloak group."""
+        org_type_to_group_mapping = {
+            OrgType.MAXIMUS_STAFF.value: GROUP_MAXIMUS_STAFF,
+            OrgType.CONTACT_CENTRE_STAFF.value: GROUP_CONTACT_CENTRE_STAFF,
+        }
         if model.membership_status.id == Status.ACTIVE.value:
             KeycloakService.join_account_holders_group(model.user.keycloak_guid)
-        elif (
-            model.membership_status.id == Status.INACTIVE.value
-            and len(MembershipModel.find_orgs_for_user(model.user.id)) == 0
-        ):
+            if model.org.type_code in org_type_to_group_mapping:
+                KeycloakService.add_user_to_group(
+                    model.user.keycloak_guid, org_type_to_group_mapping[model.org.type_code]
+                )
+        elif model.membership_status.id == Status.INACTIVE.value:
             # Check if the user has any other active org membership, if none remove from the group
-            KeycloakService.remove_from_account_holders_group(model.user.keycloak_guid)
+            if len(MembershipModel.find_orgs_for_user(model.user.id)) == 0:
+                KeycloakService.remove_from_account_holders_group(model.user.keycloak_guid)
+            if model.org.type_code in org_type_to_group_mapping:
+                KeycloakService.remove_user_from_group(
+                    model.user.keycloak_guid, org_type_to_group_mapping[model.org.type_code]
+                )
+                try:
+                    membership = Membership.get_membership_for_org_and_user(model.org.id, model.user.id)
+                    membership.deactivate_membership()
+                except BusinessException as e:
+                    logger.error(f"Error deactivating membership: {e}")
```

Comment:
seems like a really bad state if this happens, we shouldn't catch and do nothing